src/protean/core/entity.py

```diff
@@ -11,71 +11,100 @@
 
 logger = logging.getLogger('protean.core.entity')
 
+ENTITY_META_ATTRIBUTE_NAMES = ('ordering')
+
 
 class EntityBase(type):
     """
     This base metaclass sets a `_meta` attribute on the Entity to an instance
     of Meta defined in the Entity
 
-    It also sets dictionary named `declared_fields` on the `meta_` attribute.
+    It also sets dictionary named `declared_fields` on the `_meta` attribute.
     Any instances of `Field` included as attributes on either the class
     or on any of its superclasses will be include in this dictionary.
     """
 
-    @classmethod
-    def _get_declared_fields(mcs, klass, bases, attrs):
-        # Load all attributes of the class that are instances of `Field`
-        fields = []
+    def __new__(mcs, name, bases, attrs, **kwargs):
+        """Initialize Entity Classes"""
+
+        # Ensure initialization is only performed for subclasses of Entity
+        # (excluding Entity class itself).
+        parents = [b for b in bases if isinstance(b, EntityBase)]
+        if not parents:
+            return super().__new__(mcs, name, bases, attrs)
+
+        # Create the class object
+        module = attrs.pop('__module__')
+        new_attrs = {'__module__': module}
+        # Ensure we pass __classcell__ to type.__new__
+        #   Refer to https://stackoverflow.com/a/41347038/1858466 for more information
+        classcell = attrs.pop('__classcell__', None)
+        if classcell is not None:
+            new_attrs['__classcell__'] = classcell
+
+        new_class = super().__new__(mcs, name, bases, new_attrs, **kwargs)
+
+        # Gather `Meta` class/object if defined
+        attr_meta = attrs.pop('Meta', None)
+        meta = attr_meta or getattr(new_class, 'Meta', None)
+        new_class.add_to_class('_meta', EntityMeta(meta))
+
+        # Add declared fields
+        declared_fields = []
         for attr_name, attr_obj in attrs.items():
             if isinstance(attr_obj, Field):
-                # Bind the field object and append to list
-                attr_obj.bind_to_entity(klass, attr_name)
-                fields.append((attr_name, attr_obj))
+                new_class.add_to_class(attr_name, attr_obj)
```

Comment:
this needs to be outside if as we want to add to class for all attributes

Reply:
Good catch. Will change and push as part of next commit.

Reply:
Pushed in later commit